.github/workflows/sts-mock-push-to-main.yml

```diff
@@ -0,0 +1,252 @@
+name: sts-mock push to main
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "sts-mock/**"
+      - ".github/workflows/sts-mock-push-to-main.yml"
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  sts-mock-tests-and-sonar-scan:
+    name: Run tests and Sonar scan
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: sts-mock
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@9a9194f87191a7e9055e3e9b95b8cfb13023bb08 #main
+        with:
+          submodules: true
+          fetch-depth: 0
+
+      - name: Setup nodeJS v20
+        uses: actions/setup-node@26961cf329f22f6837d5f54c3efd76b480300ace #main
+        with:
+          node-version: 20
+          cache: npm
+          cache-dependency-path: sts-mock/package-lock.json
+
+      - name: Install dependencies
+        run: npm install
+
+      # Generate test coverage report for Sonar main branch analysis
+      - name: Run Tests
+        run: npm run test:unit
+
+      - name: Run infra Tests
+        run: npm run test:infra
+
+      - name: SonarCloud Scan
+        uses: sonarsource/sonarcloud-github-action@f5003fc9688ade81ce47b57a3fa97a8d3f12de4c #master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+  build-and-push-test-image-to-dev:
+    name: Build and push test image to Dev
+    needs: sts-mock-tests-and-sonar-scan
+    runs-on: ubuntu-latest
+    env:
+      STS_MOCK_DEV_TEST_IMAGE_REPOSITORY_URI: ${{ secrets.STS_MOCK_DEV_TEST_IMAGE_REPOSITORY_URI }}
+      BACKEND_API_DEV_COSIGN_KEY: ${{secrets.BACKEND_API_DEV_COSIGN_KEY}}
```

Comment:
This ENV VAR isn't used in the following steps. Is it required?

Reply:
Nope.